Hw10 security

bukazoid opened the pull request.

Commit:
new project

Commit:
users' repo and db inititalization

Commit:
couple fixes to make login possible in react.

yep, need to check example to make code more simple

Commit:
more changes

stvort commented on hw10-security/src/main/java/ru/yakimov/spring/mvc/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,64 @@
+	@Override
+	public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+		log.info("username: {}", username);
+		if (username == null || username.isEmpty()) {
+			throw new UsernameNotFoundException(username);
+		}
+		User user = dao.findByLogin(username);
```
Тут скорее надо кидать исключение когда `user == null`

Commit:
fixed exception for
username == null

Commit:
couple fixes

1 - fixed login behaviour, when login failed
2 - login error displayed (hook is used)
3 - exception on user == null
4 - default page is optional (if not chosen another page)
